### PR TITLE
Closes cppl-55

### PR DESCRIPTION
**Pull request by iodream**
Added login handler tests after the refactoring

 Added id in LoginHandler being sent as outside of the jwt token

### CODE REVIEW

**Review by MaksymTerekhovskyy — APPROVED** [review]

File: src/Server/Tests/LoginTests.cpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+#include "MockDbFacade.h"
+#include "Server/Handlers/LoginHandler.h"
+
+#include <QDebug>
+
+using ::testing::Return;
+using ::testing::_;
+
+TEST(LoginHandlerTest, USER_LOGIN_INVALID)
```
Comment (yaroslav-kernitskyi) [inline src/Server/Tests/LoginTests.cpp:15]: Strange test. Why would `GetUserByLogin` ever return you user with different login than requested? You could test if passed and returned logins are the same, but it's still looks like testing libdal and not your handler

Comment (iodream) [reply src/Server/Tests/LoginTests.cpp:16]: i wanted to simulate the password hashes not corresponding to each other. i've modified the approach slightly, is this better?

File: src/Server/Tests/LoginTests.cpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	{
+		QJsonObject json;
+		json["login"] = "Test user"; // login data is the same
+		json["password"] = "Test password hash";
+		request.json_payload = QJsonDocument(json);
+	}
+
+	auto response = handler->Handle(request);
+
+	ASSERT_EQ(response.status, Poco::Net::HTTPResponse::HTTPStatus::HTTP_OK);
```
Comment (yaroslav-kernitskyi) [inline src/Server/Tests/LoginTests.cpp:70]: I think you should also check if id returned with response is equal to `existing_user.id`

File: src/Server/Tests/LoginTests.cpp (hunk 1 above)
Comment (yaroslav-kernitskyi) [reply src/Server/Tests/LoginTests.cpp:15]: Looks fine

**Review by yaroslav-kernitskyi — APPROVED** [review]

**Pull request merged**